src/main/scala/chisel3/simulator/SimulatorAPI.scala

```diff
@@ -8,6 +8,34 @@ import java.nio.file.Files
 
 trait SimulatorAPI {
 
+  /** This object provides certain canned stimulus patterns that can be used for other
+    */
+  object Stimulus {
+
+    /** Run a simulation expecting it to safely terminate before `maxCycles`.
+      *
+      * @param maxCycles the number of cycles to ait
+      * @throws [[Exceptions.Timeout]] if `maxCycles` is reached before
+      * termination
+      */
+    def runUntilFinished[T <: Module](maxCycles: Int): (T) => Unit = { dut =>
+      AnySimulatedModule.current
+        .port(dut.clock)
+        .tick(
+          timestepsPerPhase = 1,
+          maxCycles = maxCycles,
+          inPhaseValue = 1,
+          outOfPhaseValue = 0,
+          sentinel = None,
+          checkElapsedCycleCount = (count: BigInt) => {
+            if (count == maxCycles)
+              throw new Exceptions.Timeout(maxCycles, "Expected a $finish, but none received")
+          }
+        )
+    }
```

Comment:
I think this is good--we _could_ consider doing a typeclass instead of `T <: Module`. I say that because this won't be overloadable with other types so if someone wanted to use `runUnitFinished` for a type like `BasicTester` (or some future new version of that that has a `done` output like in the new inline unit testing stuff), then they'd have to give the new function a different name. Not sure if it's worthwhile and we can also add it later in a source-compatible way so not sure it matters now. Just pointing it out.

Reply:
I switched to a type class style approach in https://github.com/chipsalliance/chisel/pull/4706/commits/d33e4614c70d4dda160797a8b39ce82bc30131e5. Let me know what you think.